src/abe_policy/policy.rs

```diff
@@ -363,8 +360,7 @@ fn generate_current_attribute_partitions(
         if !current_attr_value_per_dim.contains_key(dim) {
             // gather all the latest value for that dim
             let values = dim_properties
-                .attributes
-                .values()
+                .attributes_properties()
```

Comment:
You want to get the attributes?
```suggestion
                .attributes()
```
You can also use `values()` to follow the `HashMap` conventions (a dimension is a map from names to attributes):
```rust
                .values()
```